servers/store/src/main/proto/table.proto

```diff
@@ -43,13 +43,41 @@ message HiveDatabase {
   bytes database = 1;
 }
 
+message Dialect {
+  oneof object_type {
+    DremioDialect dremio_dialect = 1;
+    HiveDialect hive_dialect = 2;
+    SparkDialect spark_dialect = 3;
+    TrinoDialect trino_dialect = 4;
+  }
+}
+
 message SqlView {
   string sql_text = 1;
-  string dialect = 2;
+  Dialect dialect = 2;
+  string context = 3;
+  string resolved_sql_text = 4;
+  bytes schema = 5;
```

Comment:
I am not 100% sure how to store this yet. Maybe as Arrow flatbuffs if we can do a limited set of arrow deps on the server side only. Ideally i would like to prevent adding arrow as a client side dependency.